Marlin/Marlin_main.cpp

```diff
@@ -3833,6 +3833,19 @@ inline void gcode_M104() {
         setTargetHotend1(temp == 0.0 ? 0.0 : temp + duplicate_extruder_temp_offset);
     #endif
   }
+
+  // Detect if a print job has finished.
+  // When the target temperature for all extruders is zero then we must have
+  // finished printing.
+  if( print_job_start_ms != 0 ) {
```

Comment:
Minor quibble about the spacing. It should be `if (print_job_start_ms != 0)`. See https://github.com/MarlinFirmware/Marlin/wiki/Coding-Standards#spacing. (You can also leave out the `!= 0` if you feel like it and no one will complain. We aim not to be too pedantic in coding style if the meaning is fairly obvious.)